### ISSUE TITLE
WithPerTenantOptions overload allowing dependency injection?

### ISSUE THREAD

**Issue by Bouke** [Finbuckle/Finbuckle.MultiTenant#478]
In order to use dependency injected services while configuring services, you'd normally inherit from `IConfigureNamedOptions` or `IConfigureOptions` and register a singleton like this:

```cs
services.AddSingleton<IConfigureOptions<CookieAuthenticationOptions>, MyConfiguration>();
```

Inside `MyConfiguration` I can have dependencies injected:
```cs
    class MyConfiguration : IConfigureOptions<CookieAuthenticationOptions>
    {
        private IDataProtectionProvider dp;

        public GatewaySessionConfiguration(IDataProtectionProvider dp)
        {
            this.dp = dp;
        }

        public void Configure(CookieAuthenticationOptions options)
        {
            options.TicketDataFormat = TicketDataFormat(dp.CreateProtector("on purpose"));
        }
    }
```

With the current `WithPerTenantOptions<CookieAuthenticationOptions>` I cannot achieve the same. The data protector is only instantiate later on in the pipeline and I cannot grab a reference to it in `ConfigureServices`.

**Comment by AndrewTriesToCode** [comment]
Hi, I think I see what you mean -- you want to access DI from within the lambda inside `WithPerTenantOptions` right?

I will have to add this but I have a workaround that will be available before .NET 6 comes out.

**Comment by AndrewTriesToCode** [comment]
One thing you could try for now is to implement `ITenantConfigureOptions<TOptions, TTenantInfo>` and inject into the ctor there. Keep in mind it registers as singleton so any DI you do should not be for scoped services.

**Comment by Bouke** [comment]
The interface is not public, so I cannot implement it:
https://github.com/Finbuckle/Finbuckle.MultiTenant/blob/e0c88b0a12af0152f8e404ec5bbf230341a0c0d3/src/Finbuckle.MultiTenant/Options/ITenantConfigureOptions.cs#L6

**Comment by Bouke** [comment]
OT: Are you interested in a PR adding support for a `Named` variant, similar to `IConfigureNamedOptions<T>`?

**Comment by AndrewTriesToCode** [comment]
You are right it is not public! I will change it to public for the next release which will be for .Net 6 in November at the latest. 

Regarding support for names options- yes this is on my todo list and probably will come shortly after the .NET 6 release. If you want to take a stab at it and submit a PR please do so.

**Comment by AndrewTriesToCode** [comment]
I just released 6.4.2 on nuget.org which makes this a public interface. Thanks for your patience.

**Comment by AndrewTriesToCode** [comment]
Named options added to the latest release. Cheers.
